quickwit-config/src/config.rs

```diff
@@ -208,8 +193,88 @@ pub struct QuickwitConfig {
     #[serde(rename = "searcher")]
     #[serde(default)]
     pub searcher_config: SearcherConfig,
-    #[serde(rename = "storage")]
-    pub storage_config: Option<StorageConfig>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct QuickwitConfigObject {
+    pub version: usize,
+    pub cluster_id: String,
+    pub node_id: String,
+    pub listen_address: Host,
+    pub rest_listen_addr: SocketAddr,
+    pub gossip_listen_addr: SocketAddr,
+    pub grpc_listen_addr: SocketAddr,
+    pub advertise_address: Host,
+    pub gossip_advertise_addr: SocketAddr,
+    pub grpc_advertise_addr: SocketAddr,
+    pub rest_listen_port: u16,
+    pub gossip_listen_port: u16,
+    pub grpc_listen_port: u16,
+    pub peer_seeds: Vec<String>,
+    #[serde(deserialize_with = "deser_valid_uri")]
+    pub metastore_uri: Uri,
+    #[serde(deserialize_with = "deser_valid_uri")]
+    pub default_index_root_uri: Uri,
+    pub data_dir_path: PathBuf,
+    pub indexer_config: IndexerConfig,
+    pub searcher_config: SearcherConfig,
+}
+
+impl QuickwitConfigObject {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        validate_identifier("Cluster ID", &self.cluster_id)?;
+        validate_identifier("Node ID", &self.node_id)?;
+
+        if self.cluster_id == DEFAULT_CLUSTER_ID {
+            warn!(
+                cluster_id = DEFAULT_CLUSTER_ID,
+                "Cluster ID is not set, falling back to default value."
+            );
+        }
+        if self.peer_seeds.is_empty() {
+            warn!("Seed list is empty.");
+        }
+        let data_dir_uri = Uri::try_new(&self.data_dir_path.to_string_lossy())?;
+
+        if !data_dir_uri.protocol().is_file() {
+            bail!("Data dir must be located on local file system")
+        }
+        if !self.data_dir_path.exists() {
+            bail!(
+                "Data dir `{}` does not exist.",
+                self.data_dir_path.display()
+            );
+        }
+        Ok(())
+    }
+
+    /// Returns the list of peer seed addresses. The addresses MUST NOT be resolved. Otherwise, the
+    /// DNS-based discovery mechanism implemented in Chitchat will not work correctly.
+    pub async fn peer_seed_addrs(&self) -> anyhow::Result<Vec<String>> {
```

Comment:
Why have you kept those behind an accessor?

Reply:
It was not working properly when I tried to generate addresses on config load time. It couldn't resolve the addresses of `peer_seeds` (probably due to DNS stuff but I couldn't understand why).

Reply:
Let's investigate why this didn't work.